pkg/metrics/api/api.go

```diff
@@ -0,0 +1,456 @@
+	// ClusterInstallationModificationBootstrapJobName is the variable that stores the metric name for the modification of bootstrap manifests details.
+	ClusterInstallationModificationBootstrapJobName = "cluster_installation_modification_bootstrap_ignition"
+	// ClusterInstallationModificationMasterJobName is the variable that stores the metric name for the modification of master manifests details.
+	ClusterInstallationModificationMasterJobName = "cluster_installation_modification_master_ignition"
+	// ClusterInstallationModificationWorkerJobName is the variable that stores the metric name for the modification of worker manifests details.
+	ClusterInstallationModificationWorkerJobName = "cluster_installation_modification_worker_manifest"
+)
+
+var (
+	// FileCategories is a map of asset names to the categories of the metrics that it belongs to.
+	FileCategories = make(map[string]string)
+	// metricRegistry holds all the MetricBuilder objects for all the metrics that have been designed
+	// and ready for extraction from the installer.
+	metricRegistry = make(map[string]builder.MetricBuilder)
+
+	// CreateCommandMetricName is the map of the targets to the respective metric names. Installer can
+	// look this up for the correct metric name given a target.
+	CreateCommandMetricName = make(map[string]string)
+
+	// optsRegistry holds all the information that every metric needs to create a metric builder.
+	optsRegistry = make(map[string]builder.MetricOpts)
+
+	// CurrentInvocationContext is used to store the current Invocation metric name used for building context.
+	// Invocation data information is gathered from different parts of the installer and it gets hard for the
+	// installer to keep track of the current command and target that is being run. Hence, the current invocation
+	// metric name is stored here in this variable.
+	CurrentInvocationContext string
+
+	// enableMetrics holds the user preference for sending metrics. If the user chooses not to send, this will hold
+	// the value false.
+	enableMetrics bool
+
+	// prometheusURL is the standard URL to send metrics to. The user can override this information to any URL they choose.
+	prometheusURL = "https://localhost:9092/"
+
+	// pushClient holds the URL, jobname and the http client information to send metrics.
+	pushClient pushclient.PushClient
+)
+
+// Initialize holds all the initial setup information like creating all the metrics that are allowed to
+// be extracted, checking for user input about disabling metrics, URL and for creating the Push Client.
+// Must be run before the metircs are collected.
+func Initialize() {
+	enableMetrics = false
+
+	// if _, ok := os.LookupEnv("OPENSHIFT_INSTALL_DISABLE_METRICS"); ok {
+	// 	enableMetrics = true
+	// }
+
+	if value, ok := os.LookupEnv("OPENSHIFT_INSTALL_METRICS_ENDPOINT"); ok {
+		prometheusURL = value
+		enableMetrics = true
+	}
+	initializeModificationCategories()
+	initializeCreateCommandMetricName()
+	initializeOptsForMetrics()
```

Comment:
As these are initializing package level variables, I don't really understand why we need these--rather I think we should just initialize the package level variables with literals. For example:

```go
optsRegistry = map[string]builder.MetricOpts{
     ClusterInstallationCreateJobName : builder.MetricOpts{
		Name:   ClusterInstallationCreateJobName,
		Labels: []string{"result", "returnCode", "platform", "os", "version"},
		Desc: "This metric keeps track of the count of the number of times " +
			"the user ran create cluster command in the given OS " +
			"took the time that is lesser than or equal to the value in the duration label.",
		Buckets:    prometheus.LinearBuckets(15, 5, int(math.Ceil(60/5))+1),
		metricType: builder.Histogram,
	},
        ClusterInstallationIgnitionJobName : builder.MetricOpts{
		Name:   ClusterInstallationIgnitionJobName,
 ...
}
```